### PR TITLE
Implement Page.Events.Dialog

### LINKED ISSUE #18: Implement Page.Events.Dialog tests
https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/test/test.js#L1584

### PR DESCRIPTION
**Pull request by kblok**
closes #18

### CODE REVIEW

File: lib/PuppeteerSharp/Messaging/PageJavascriptDialogOpeningMessageResponse.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using Newtonsoft.Json;
+
+namespace PuppeteerSharp.Messaging
```
Comment (kblok) [inline lib/PuppeteerSharp/Messaging/PageJavascriptDialogOpeningMessageResponse.cs]: @Meir017 I'm introducing this `Messaging` namespace so we can start working on #97, little by little

**Comment by kblok** [comment]
Ready to review @Meir017

File: lib/PuppeteerSharp/DialogType.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿namespace PuppeteerSharp
+{
+    public enum DialogType
```
Comment (Meir017) [inline lib/PuppeteerSharp/DialogType.cs]: we should not have this and the class `Dialog.Type`. This should replace the constants

Comment (kblok) [reply lib/PuppeteerSharp/DialogType.cs]: Agreed, How about now?

File: lib/PuppeteerSharp/Page.cs [hunk 3]
```diff
@@ -714,8 +716,33 @@ private void HandleException(string exceptionDetails)
         {
         }
 
-        private void OnDialog(MessageEventArgs e)
+        private void OnDialog(PageJavascriptDialogOpeningMessageResponse message)
         {
+
+            DialogType dialogType;
+            if (message.Type == "alert")
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: maybe create a dictionary to map the type to its string representation to clean this up?

**Comment by kblok** [comment]
How about now @Meir017?

**Pull request merged**